### ISSUE TITLE
Printing exceptions in the REPL is sometimes slow

### ISSUE THREAD

**Issue by maxrothman** [BetterThanTomorrow/calva#1599]
Possibly related: https://github.com/BetterThanTomorrow/calva/issues/1051
Discussion threads: [first](https://clojurians.slack.com/archives/CBE668G4R/p1647444183989439), [second](https://clojurians.slack.com/archives/CBE668G4R/p1642803700114400)

I have two computers on which I use Calva. On one of them, when an exception is raised as a result of executing a form in the REPL, it takes ~3-4 seconds to print the exception message and traceback prompt. On the other, exception messages are printed instantaneously.

Specs on the offending computer:
- MacOS 10.15.7
- VSCode 1.65.2
- Calva 2.0.254

How to replicate: 
- `lein new app replicating`
- Open `core.clj` in calva
- Jack in, select "Leiningen"
- Run "Calva: Load Current File and Dependencies"
- Add `(throw (ex-info "hi" {:a 1}))`
- Calva: Evaluate Current Form on the above line

Relevant config
```json
    "[clojure]": {
        "editor.wordSeparators": "`~!@#$%^&*()=+[{]}\\|;:'\",.<>/?",
        "editor.tabSize": 2,
        "editor.guides.bracketPairs": "active",
        "editor.bracketPairColorization.enabled": true,
    },
    "calva.highlight.enableBracketColors": false,
    "calva.paredit.defaultKeyMap": "strict",
    "calva.showCalvaSaysOnStart": false,
    "calva.prettyPrintingOptions": {
        "enabled": true,
        "printEngine": "pprint",
        "width": 70,
        "maxLength": 50
    },

    "calva.highlight.matchedBracketStyle": {
        "borderWidth": "1px",
        "borderStyle": "solid",
        "borderColor": "#b3b3b3"
    }
```

One more note: some of the time when an exception is thrown, I also see a small error popup in the bottom right of the screen that says "n.filter is not a function". This seems to happen more often when the error is a CompilerException, but it's inconsistent.

**Comment by bpringe** [comment]
Here's an issue with repro steps for the n.filter issue: https://github.com/BetterThanTomorrow/calva/issues/1567.

**Comment by bpringe** [comment]
Have you tried disabling other extensions to see if that helps? Is there any difference in VS Code or Calva versions between the offending computer and the one where the slowness is not present?

It seems you have reproduced the issue in a new project, so at least the issue being project-specific seems ruled out.

**Comment by maxrothman** [comment]
Whoops, sorry for the close! Was in the middle of writing this comment and accidentally hit a button.

Sorry for the delay in getting back to this!

> Have you tried disabling other extensions to see if that helps?

I tried disabling all extensions other than Calva, there was no change in behavior.

> Is there any difference in VS Code or Calva versions between the offending computer and the one where the slowness is not present?

On the problematic computer:
- OS: MacOS 10.15.7
- VSCode: 1.66.0
- Calva: 2.0.264

On the working computer:
- OS: mac 12.3.1
- VSCode: 1.66.0
- Calva: 2.0.264

**Comment by maxrothman** [comment]
Did a little more experimenting, and I think I found the cause of the slowness on that one computer! On the affected computer, `~/.lein/profiles.clj` looked like this:
```clojure
{:user {:plugins [[cider/cider-nrepl "0.24.0"]]}
```
Changing `profiles.clj` to an empty map solved the issue.

I don't know why that stuff was in my `profiles.clj`, I probably set it up when I was first getting into Clojure and copied it out of some tutorial somewhere.

I also remembered that I'd noticed that one of my projects at work experiences this slowness issue. In that project, we start a REPL in a docker container and connect to it using Calva. I checked the REPL start command, and it **also** specified a pre-0.27-* version of cider-nrepl. Exception reporting is still a little slow, but it's down to ~2s from like ~30s, which is a huge improvement. Everything is a little laggier with the dockerized REPL anyways, so I suspect the exception reporting is now hitting the same bottleneck everything else is.

Long story short, the issue appears to be with using a pre-0.27.* version of cider-nrepl. Updating to a post-0.27-* version of cider-nrepl appears to resolve the issue.

**Comment by PEZ** [comment]
Now I feel slow... I remember this issue being discussed and then fixed in cider-nrepl.

I take it you are not using Jack-in, @maxrothman ? Or is the user profile winning over the command line args?

**Comment by maxrothman** [comment]
In the projects except for the one with the dockerized REPL, I am using jack-in. I suspect the profile is winning over the args? Though that itself is a bit surprising.

**Comment by bpringe** [comment]
I seem to remember something about lein resolving dependencies in such a way that the last-seen version of the same dep on the classpath is used (or something), rather than the latest version, which isn't the case with the Clojure CLI. I could be misremembering though, and I don't know if that would even be the issue here.

**Comment by PEZ** [comment]
To me it makes sense to let command line args override anything in any config. I remember asking about this on the Leiningen repo quite long ago, but I don't remember the answer to be surprising. Maybe I didn't understand the answer or maybe I didn't even get one. I'll have to dig a bit...

**Comment by bpringe** [comment]
> To me it makes sense to let command line args override anything in any config

I agree. Maybe they do always override with lein.

**Comment by bpringe** [comment]
Closing since it was found to be a dep issue.
